src/ruptures/costs/costml.py

```diff
@@ -40,6 +41,8 @@ def fit(self, signal) -> "CostMl":
             self.metric = inv(covar.reshape(1, 1) if covar.size == 1 else covar)
 
         self.gram = s_.dot(self.metric).dot(s_.T)
+        self.signal = deepcopy(s_)
+
```

Comment:
I don't think we need a deepcopy here. The signal is never changed inplace.

`self.signal = s_` is enough

Reply:
Done !